FlowCrypt/Functionality/Mail Provider/MailProvider.swift

```diff
@@ -81,12 +76,12 @@ final class MailProvider {
         resolveService(of: MessagesThreadOperationsProvider?.self)
     }
 
-    private init(
-        currentAuthType: @autoclosure @escaping () -> (AuthType?),
-        services: [MailServiceProvider]
+    init(
+        currentAuthType: AuthType,
+        currentUser: User
     ) {
         self.currentAuthType = currentAuthType
-        self.services = services
+        self.services = MailServiceProviderFactory.services(user: currentUser)
     }
```

Comment:
I was a bit worried about this - originally dynamic as a closure, now user is directly provided. Need to scrutinize more. But app seems to work the same.